feat(flags): Consolidate feature flags pull requests (#12005)
getsentry/sentry-docs: docs/organization/integrations/feature-flag/launchdarkly/index.mdx

Reviewer: This file could talk about eval tracking in issues, and linking to the list of provider/language list of setup instructions for that too.

---
title: LaunchDarkly
sidebar_order: 1
description: Learn about Sentry's LaunchDarkly integrations.
---

## Evaluation Tracking

Sentry can track flag evaluations as they happen within your application.  Flag evaluations will appear in the "Feature Flag" section of Issue Details page as a table, with "suspect" flag predictions highlighted in yellow. Learn more about how to interact with feature flag insights within the Sentry UI by reading the [Issue Details page documentation](/product/issues/issue-details/#feature-flags).

### Set Up Evaluation Tracking

To set up evaluation tracking visit one of our supported languages documentation page:

> * [JavaScript](/platforms/javascript/configuration/integrations/launchdarkly/)

Reviewer: ```suggestion
* [JavaScript](/platforms/javascript/feature-flags/)
```

* [Python](/platforms/python/integrations/launchdarkly/)

## Change Tracking

Sentry can track changes to feature flag definitions and report suspicious feature flag edits.

### Install and Configure

To enable the change-tracking integration, go to the **Issue Details** page and click the "Set Up Integration" button on the feature flag table. This will trigger a flyout, which will guide you through the two-step process.

![Integration UI](./img/ff-integration-ui.png)

> Under the "Integrate Feature Flag Service" heading a Sentry webhook URL has been provided. The Sentry URL must be registered with LaunchDarkly as a webhook integration within their [webhook integration UI](https://app.launchdarkly.com/settings/integrations/webhooks/new). Paste the Sentry URL into the input box labeled "URL" within the LaunchDarkly webhook registration UI.

Reviewer 2: ```suggestion
Under the "Integrate Feature Flag Service" section, you'll find a Sentry webhook URL and a link to LaunchDarkly's webhook registration interface.

1. **Register the Webhook:** Paste the Sentry webhook URL into the "URL" field in the LaunchDarkly [webhook integration UI](https://app.launchdarkly.com/settings/integrations/webhooks/new).
```

description: Learn about Sentry's LaunchDarkly integration. LaunchDarkly enables organizations to use Sentry errors as a metric in their LaunchDarkly experiments.
---

## Metrics Integration

This integration is maintained and supported by the company that created it. For more details or questions, feel free to contact rrosefigura@launchdarkly.com.

> ## Install and Configure

Reviewer: should probably indent this as it relates to the heading above only.

1. Navigate to **Settings > Integrations > LaunchDarkly**

2. Follow the full [LaunchDarkly installation instructions](https://docs.launchdarkly.com/integrations/sentry).

## Change Tracking

Sentry can track changes to feature flag definitions and report suspicious feature flag edits. To enable the change tracking integration click the "Set Up Integration" button on the feature flag table contained within the issue details page. This will trigger a fly out which will guide you through the two step process.

![Integration UI](./img/ff-integration-ui.png)

Under the "Integrate Feature Flag Service" heading a Sentry webhook URL has been provided. Additionally, a link to LaunchDarkly's webhook registration interface has been provided. Paste the Sentry webhook URL into the input box labeled "URL" within the LaunchDarkly webhook registration UI.

> Next a signing secret must be created. Check the box that says "Sign this webhook". This will reveal an input element which contains your signing secret. Copy and paste this signing secret into the input box labeled "Secret" on the Sentry fly out. Click "Save Secret" in the Sentry UI, click "Save Settings" in the LaunchDarkly UI, and you're done. Sentry will now accept and authenticate all inbound hooks to your organization's feature flag webhook endpoint.

Reviewer: needs an "Install and Configure" heading like the above Metrics Integration?

1. Navigate to **Settings > Integrations > LaunchDarkly**

2. Follow the full [LaunchDarkly installation instructions](https://docs.launchdarkly.com/integrations/sentry).

## Evaluation Tracking

Sentry can track flag evaluations as they happen within your application.  Flag evaluations will appear in the "Feature Flag" section of Issue Details page as a table, with "suspect" flag predictions highlighted in yellow. Learn more about how to interact with feature flag insights within the Sentry UI by reading the [Issue Details page documentation](/product/issues/issue-details/#feature-flags).

### Set Up Evaluation Tracking

To set up evaluation tracking visit one of our supported languages documentation page:

> * [JavaScript](/platforms/javascript/feature-flags/)
> * [Python](/platforms/python/feature-flags/)

Reviewer: ```suggestion
* [JavaScript](/platforms/javascript/configuration/integrations/launchdarkly/)
* [Python](/platforms/python/integrations/launchdarkly/)
```

## Change Tracking

Sentry can track changes to feature flag definitions and report suspicious feature flag edits.

### Install and Configure

> To enable the change tracking integration click the "Set Up Integration" button on the feature flag table contained within the issue details page. This will trigger a fly out which will guide you through the two step process.

Reviewer: ```suggestion
To enable the change tracking integration go to the issue details page and click the "Set Up Integration" button on the feature flag table. This will trigger a fly out which will guide you through the two step process.
```

put sentence in order of operation, visit then click.

![Integration UI](./img/ff-integration-ui.png)

> Under the "Integrate Feature Flag Service" heading a Sentry webhook URL has been provided. Additionally, a link to LaunchDarkly's webhook registration interface has been provided. Paste the Sentry webhook URL into the input box labeled "URL" within the LaunchDarkly webhook registration UI.

Reviewer: I'm noticing now that the copywriting in the modal is very generic, but the link is for launchdarkly.com. We might want to change that link from "feature flag service" to clearly say "launch darkly", but that's code in another repo

Reviewer: There is an image in another PR that didn't make it into here: https://github.com/getsentry/sentry-docs/pull/11749/files#diff-f09a4ba0a7ec1f45114b4bed5b636d218f53f9b9a5187b861f20d9a97140b970

The "Generic Trigger" screenshot from LD's website. that could be an asset to bring over and improve the setup steps here. For me, having the structure set (which we do!) is enough for now, we can iterate on the specific steps within this page later.

Author: Added a URL to the webhook page.  Took a note about the copywriting will alert michelle.
